src/app.js

```diff
@@ -22,9 +25,22 @@ app.use(bodyParser.json({ type: 'application/vnd.api+json' }));
 app.use(methodOverride('X-HTTP-Method-Override'));
 app.use(passport.initialize());
 
-app.use('/users', usersRouter);
+const userPermissionCheck = [
```

Comment:
This is not a proper place for this code.

make an applyMiddleware function in middlewares/index.js that will accept app instance and decorate it with all of the custom middleware you need.

You can then import it here and use it.

Then create a app.use('/v1, v1routes); (v1routes will be exported from /routes/index.js)

This will give you flexibility in API versioning (read about it when you have the time)